Tests/UnitTests/RequestCoordinatorTests.swift

```diff
@@ -245,5 +245,88 @@ class RequestCoordinatorTests: XCTestCase {
         
         self.wait(for: [expectation], timeout: 2.0)
     }
+    
+    func testEnqueueRetriableRequest_backsOff_succeeds() {
+        // Arrange
+        let expectation = self.expectation(description: "500 -> 500 succeeds")
+        let url = URL(string: "https://usebutton.com")!
+        let request = URLRequest(url: url)
+        let expectedData = Data()
+        
+        // Act
+        coordinator.enqueueRetriableRequest(request: request, attempt: 0, maxRetries: 3, retryIntervalInMS: 10) { data, error in
+            
+            // Assert
+            XCTAssertNil(error)
+            XCTAssertEqual(data, expectedData)
+            XCTAssertEqual(self.session.allDataTasks.count, 2)
+            
+            expectation.fulfill()
+        }
+        
+        session.lastDataTask?.completion(nil, nil, nil)
+        XCTAssertEqual(self.session.allDataTasks.count, 1)
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.020) {
+            let successfulResponse = HTTPURLResponse(url: url, statusCode: 200, httpVersion: nil, headerFields: nil)
+            self.session.lastDataTask?.completion(expectedData, successfulResponse, nil)
+        }
+        
+        self.wait(for: [expectation], timeout: 2.0)
+    }
+    
+    func testEnqueueRetriableRequest_backsOff_multipleTimes_succeeds() {
```

Comment:
This test looks very fragile. Why are we using "sleep" to test here? Why not test that calling the completion handler updates the inputs to the next attempt. Call the code under test, capture the completion block, invoke it with an error that causes retry, assert that it updated the number of retries and the next retryIntervalInMS.

I'm also unclear where we're testing that a 200, 400 does _not_ trigger a retry.